src/cpu/flexcpu/simple_dataflow_cpu.cc

```diff
@@ -76,8 +84,23 @@ SimpleDataflowCPU::activateContext(ThreadID tid)
 void
 SimpleDataflowCPU::attemptAllFetchReqs()
 {
+    if (clockedInstFetch) {
+        if (curTick() != clockEdge()) {
+            if (fetchAttemptScheduled) {
+                return;
+            }
+
+            callAtClockEdge([this] { attemptAllFetchReqs(); });
```

Comment:
Hmm my thought when doing it this way was just so that control flow was internal to the functions where it mattered.

It's probably more efficient not to recreate events where no new lambdas are necessary to capture parameters or anything.

Reply:
Should probably do this call after the call to if (_instPort.receivedPushback) actually, since you wouldn't really want to schedule a call anyway if the recvReqRetry call is going to schedule one anyway...